Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs

```diff
@@ -72,28 +69,46 @@ public async Task StopSearch()
             }
         }
         
-        public async Task StartRssiPolling(String btguid, Func<int, int> updateRssi)
+        public void StartRssiPolling(String btguid, Action<int> updateRssi, Action connected = null, Action disconnected = null)
         {
-            try
+            StopRssiPolling();
+            rssiCancel = new CancellationTokenSource();
+            var token = rssiCancel.Token;
+            Task.Run(async () =>
             {
-                IDevice device = await adapter.ConnectToKnownDeviceAsync(Guid.Parse(btguid));
-
-                rssiPollingTimer = new Timer(async (o) => {
-
-                    await device.UpdateRssiAsync();
-                    updateRssi.Invoke(device.Rssi);
-
-                }, "", 0, RssiPollInterval);
-            }
-            catch (Exception e)
-            {
-                Console.WriteLine(e.ToString());
-            }
+                while (!token.IsCancellationRequested)
+                {
+                    try
+                    {
+                        IDevice device = await adapter.ConnectToKnownDeviceAsync(Guid.Parse(btguid));
+
+                        if (!(connected is null)) connected.Invoke();
+
+                        try
+                        {
+                            while((!token.IsCancellationRequested) && device.State == DeviceState.Connected)
+                            {
+                                await device.UpdateRssiAsync();
+                                updateRssi.Invoke(device.Rssi);
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            Console.WriteLine(e.ToString());
+                        }
+                        if (!(disconnected is null)) disconnected.Invoke();
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.ToString());
+                    }
+                }
```

Comment:
```suggestion
                while (!token.IsCancellationRequested)
                {
                    try
                    {
                        IDevice device = await adapter.ConnectToKnownDeviceAsync(Guid.Parse(btguid));

                        if (!(connected is null)) connected.Invoke();

                        try
                        {
                            while((!token.IsCancellationRequested) && device.State == DeviceState.Connected)
                            {
                                await device.UpdateRssiAsync();
                                updateRssi.Invoke(device.Rssi);
                                await Task.Delay(100);
                            }
                        }
                        catch (Exception e)
                        {
                            Console.WriteLine(e.ToString());
                        }
                        if (!(disconnected is null)) disconnected.Invoke();
                    }
                    catch (Exception e)
                    {
                        Console.WriteLine(e.ToString());
                    }
                }
```
Including this delay also improves the UX, because the circle size is not varying constantly. 
The length of the delay is up to decision and testing.